build_tools/third_party/cuda/BUILD.template

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("@iree_core//build_tools/embed_data:build_defs.bzl", "c_embed_data")
+
+package(default_visibility = ["//visibility:public"])
+
+# The repository rule will perform some substitutions that we use to
+# customize the build based on the actually found toolkit.
+ENABLED = %ENABLED%
+LIBDEVICE_REL_PATH = "%LIBDEVICE_REL_PATH%"
+
+# Note that this BUILD file is overlaid on top of a partial CUDA SDK directory
+# as symlinked by the cuda_auto_configure repository rule in
+# build_tools/bazel/workspace.bzl
+
+# This flag will be True if the workspace repository rules detected a usable
+# SDK. config_settings in the project can join over this in order to
+# conditionally enable CUDA features based on pressence of the SDK.
+bool_flag(
+    name = "enabled",
+    build_setting_default = ENABLED,
+)
+
+# If the SDK was not found, these targets will be missing inputs and will
+# error if used. This will happen if someone tries to build them directly
+# or transitively.
+
+cc_library(
+    name = "headers",
+    hdrs = glob([
+        "include/**/*.h",
+        "include/**/*.hpp",
+    ]),
+    includes = ["include"],
+)
+
+c_embed_data(
+    name = "libdevice_embedded",
+    srcs = [
+        LIBDEVICE_REL_PATH,
+    ],
+    c_file_output = "iree_cuda/libdevice_embedded.c",
+    flatten = True,
+    generator = "@iree_core//build_tools/embed_data:generate_embed_data",
```

Comment:
This gave me no end up trouble. No... because this is a different workspace and it needs to get back to the main one for that dumb rule. I am open to suggestions.

Reply:
Ah yes, I have encountered such problems before. I think it *should* work if you left in the `clean_dep` and didn't specify this here. Let me play with it

Reply:
Ok, so it appears that the `Label` syntax is supposed to make things work for other repos, but if that other repo is one that the main repo instantiated, then everything breaks again because of course it does. I think the best fix is to inject the IREE repo name, as we already do with the `configure_iree_submodule_deps` macro. Sent a PR to this branch: https://github.com/iree-org/iree/pull/11666